addons/account/wizard/account_payment_register.py

```diff
@@ -519,41 +529,90 @@ def _compute_show_require_partner_bank(self):
+            for move in moves:
+                if early_payment_discount and move._is_eligible_for_early_payment_discount(move.currency_id, self.payment_date):
+                    amount -= move.direction_sign * move.invoice_payment_term_id._get_amount_due_after_discount(move.amount_total, move.amount_tax)
+                    mode = 'early_payment'
+                else:
+                    for aml in amls.filtered(lambda l: l.move_id.id == move.id):
+                        amount += aml.amount_residual_currency
+            self.next_payable_amount = abs(amount)
+            return abs(amount), mode
+
+        if self.installments_mode == 'full' or total_amount:
+            # We don't handle installments if we need to compute the total amount
+            if early_payment_discount and moves._is_eligible_for_early_payment_discount(moves.currency_id, self.payment_date):
+                amount -= moves.direction_sign * moves.invoice_payment_term_id._get_amount_due_after_discount(moves.amount_total, moves.amount_tax)
                 mode = 'early_payment'
             else:
-                for aml in batch_result['lines'].filtered(lambda l: l.move_id.id == move.id):
+                for aml in amls:
                     amount += aml.amount_residual_currency
-        return abs(amount), mode
 
-    def _get_total_amount_in_wizard_currency_to_full_reconcile(self, batch_result, early_payment_discount=True):
-        """ Compute the total amount needed in the currency of the wizard to fully reconcile the batch of journal
-        items passed as parameter.
+            if not total_amount:
+                labels = {line.name or line.move_id.ref or line.move_id.name for line in amls}
+                self.communication = " ".join(sorted(labels))
+                self.next_payable_amount = abs(amount)
+                if len(amls) > 1:
+                    self.can_group_payments = True
+            return abs(amount), mode
+
+        open_installments = amls.filtered(lambda l: l.amount_residual_currency != 0).sorted(key='date_maturity')
+        amls_to_pay = []
+        for aml in open_installments:
+            if aml.date_maturity < self.payment_date:
+                amount += aml.amount_residual_currency
+                amls_to_pay.append(aml)
+                mode = 'overdue'
+            elif mode != 'overdue':
+                amount += aml.amount_residual_currency
+                amls_to_pay.append(aml)
+                mode = 'next'
+                break
+            else:
+                break
+
+        if self.can_group_payments and len(amls_to_pay) == 1:
+            self.can_group_payments = False
+
+        labels = {line.name or line.move_id.ref or line.move_id.name for line in amls_to_pay}
+        self.communication = " ".join(sorted(labels))
+        self.installments_mode = mode if len(open_installments) > 1 else False
+        self.next_payable_amount = abs(amount)
```

Comment:
By design, it's so weird to assign value in a `_get` function called from others `compute` methods. You can't do that! You are assigning fields outside of a compute method's scope.

Reply:
![yes-we-can-obama](https://github.com/user-attachments/assets/c425a898-fd2b-4098-9484-1ce412e21eb7)

Just kidding 😆 No, you're right, it's not really clean this way. I wanted to save some computations, but it's probably not the best way. I updated the code now to use `compute` functions for all these fields.